includes/admin/class.llms.admin.builder.php

```diff
@@ -1096,6 +1099,14 @@ private static function update_questions( $questions, $parent ) {
 
 				$question = $parent->questions()->get_question( $question_id );
 
+				/**
```

Comment:
Without testing this I'm not 100% positive but if we have a "default" value then we don't actually need this bit of code, right?

What I mean is doing both this and then also setting a fallback value when the value isn't set is a bit redundant.

I think the idea is that this would "fix" the problem from happening in the future and setting a default would help with existing items. However I don't think we technically need both sets of code.

Since having a default value will protect against malformed API requests and already malformed questions I'm partial to that one if we had to choose.

But maybe I'm missing something?

Reply:
@thomasplevy 

No and yes.
No: you're not missing anything from what I said - sorry.
Yes: because I used in many places "set" and "not set" concepts but I HAD to say that the culprit `question_type` - according to the course imported - IS SET but IS EMPTY.
So setting the `$property_defaults` doesn't work - that's a fallback for non set property.

The redundancy I added was just to provide a way to normalize data with minimum effort.
Do you think I can remove it then?

I think I can also add the default property as you suggest to be more strict...

Reply:
hmmm... that's weird then, i see the reasoning and in that case this (as it is without the default properties) should be good.

I know we don't know but I'm really curious how these questions are ending up with no type in the database... Seems weird